src/contentPopups/utils/messageValidation.js

```diff
@@ -0,0 +1,24 @@
+
+export const isMessageOriginValid = (msg, eventOrigin) => {
+    if (!msg?.data?.url) {
+        return true
```

Comment:
Think this could be a potential security issue:. This is dangerous in a postMessage or cross-window communication context. An attacker can simply omit the url field and bypass validation entirely.

```js
if (!msg?.data?.url) {
    return false
}
```